Simplified/MyBooks/NYPLMyBooksDownloadCenter+AudiobooksDownload.swift

```diff
@@ -9,8 +9,54 @@
 import Foundation
 import NYPLAudiobookToolkit
 
+extension NYPLMyBooksDownloadCenter {
+  /// Create an audiobook object and add to download queue
+  @objc(downloadAudiobookForBook:)
+  func downloadAudiobook(for book: NYPLBook?) {
+    guard let book = book,
+          let url = self.fileURL(forBookIndentifier: book.identifier) else {
+      return
+    }
+    
+    AudiobookManifestAdapter.transformManifestToDictionary(for: book,
+                                                              fileURL: url) { json, decryptor, error in
+      if error == .none {
+        guard let audiobook = AudiobookFactory.audiobook(json, decryptor: decryptor) else {
+          Log.info(#file, "Audiobook initiate failed for book id - \(book.identifier)")
+          // TODO: Handle Error
+          self.broadcastUpdate(book.identifier)
+          return
+        }
+        
+        let metadata = AudiobookMetadata(title: book.title, authors: [(book.authors ?? "")])
+        let manager = DefaultAudiobookManager(metadata: metadata, audiobook: audiobook)
+        
+        NYPLBookRegistry.shared().setStateWithCode(NYPLBookState.Downloading.rawValue, forIdentifier: book.identifier)
+        
+        self.downloadProgressDidUpdate(to: Double(manager.networkService.downloadProgress),
+                                       forBookIdentifier: book.identifier)
+        
+        self.audiobookDownloader.downloadAudiobook(for: book.identifier, audiobookManager: manager)
+      } else {
+        Log.info(#file, "Audiobook manifest corrupted/unsupported")
```

Comment:
info --> error. Can you also add the book.loggableString to the logged statement?